### PR TITLE
Update to build with github actions / ruby 3.0 / rails 6.1

### PR DESCRIPTION
**Pull request by jmortlock**
### WHY

Switch build to github actions
Add gem publish
Build on Ruby 3
Build against Rails 6.1
Remove Rails 5.2 build
Add dependabot

### CODE REVIEW

**Review by tlclink — APPROVED** [review]

File: table_creator.gemspec [hunk 1]
```diff
@@ -19,15 +19,16 @@ Gem::Specification.new do |spec|
...
   spec.add_dependency 'activesupport' # Hash#except, blank?, etc.
 
-  spec.add_development_dependency "bundler", "~> 1.3"
+  spec.add_development_dependency "bundler", "~> 2"
   spec.add_development_dependency "rake"
   spec.add_development_dependency 'rspec'
   spec.add_development_dependency 'coverage-kit'
   spec.add_development_dependency 'simplecov-rcov'
   spec.add_development_dependency 'coveralls'
-  spec.add_development_dependency 'travis'
```
Comment (mnoack) [inline table_creator.gemspec:32]: Are we deleting travis? Should we also remove .travis.yml?

**Comment by mnoack** [comment]
Looks good, I see github actions are quite nice.

**Comment by jmortlock** [comment]
> Looks good, I see github actions are quite nice.

Travis forced our hand a bit removing the free tier.

But github actions are quite nice and gives me an excuse to add a publish stage and get everything building with the latest rubies / rails

File: table_creator.gemspec (hunk 1 above)
Comment (jmortlock) [reply table_creator.gemspec:32]: good catch :)

**Review by sam-wan — APPROVED** [review]

**Pull request merged**